tests/upgrade/basic/task.yaml

```diff
@@ -45,8 +45,12 @@ execute: |
     snap list | grep test-snapd-tools
     test-snapd-tools.echo Hello | grep Hello
     test-snapd-tools.env | grep SNAP_NAME=test-snapd-tools
-    echo Hello > /var/tmp/myevil.txt
-    test-snapd-tools.cat /var/tmp/myevil.txt && exit 1 || true
+
+    # only test if confinement works and we actually have apparmor available
+    if [ -e /sys/kernel/security/apparmor ]; then
```

Comment:
Can you please add a note that says this should migrate to the internal command that queries forced devmode.